docs/dali_dynamic.rst

```diff
@@ -0,0 +1,90 @@
+DALI Dynamic
+========
+
+Overview
+--------
+
+**DALI Dynamic** extends NVIDIA DALI by introducing an imperative execution model with lazy evaluation.  
+It complements the existing graph-based pipeline execution and is intended for rapid prototyping, 
+easier debugging, and seamless integration into existing Python workflows.
+
+Key features include:
+
+- **Lazy execution**  
+  In the original DALI execution model, operators were defined and executed within a static 
+  pipeline graph. This often resulted in a steep learning curve, complex debugging, and 
+  limited error visibility. DALI Dynamic introduces imperative programming with lazy operator 
+  execution, aligning DALI more closely with standard Python workflows.  
+
+- **Minimal performance overhead**  
+  DALI Dynamic is designed to deliver performance that is close to graph-based pipelines, incurring 
+  only marginal overhead.  
+
+- **Batch processing support**  
+  Batch processing remains a core concept in DALI. DALI Dynamic preserves this functionality and 
+  introduces a dedicated API for batch-oriented workflows.  
+
+- **Framework interoperability**  
+  DALI Dynamic provides type conversion support for major deep learning frameworks, including 
+  PyTorch, CuPy, JAX, and Numba.
+
+.. note::
+
+   DALI Dynamic does not replace the graph-based execution model. Instead, it provides 
+   an alternative interface for defined use cases. Prototyping and development 
+   can be performed in DALI Dynamic, with subsequent transition to pipelines for 
+   high-performance, production scenarios.
+
+
+How it works
+------------
+
+DALI traditionally relies on explicitly defined pipelines, where data processing 
+is specified as a computation graph. A typical usage pattern involves defining 
+a decorated function with ``@pipeline_def``, building the pipeline, and executing 
```

Comment:
Just food for thought - this way it already reads pretty good for a seasoned DALI developer, but should we introduce the gist of DALI dynamic first, instead of showing another example of DALI Pipeline?

Also I would suggest, to compare the new flow to a regular imperative Python library like numpy (or cupy, torch or whatever), rather than focusing on the pipeline introduction.